CSS3.sublime-syntax

```diff
@@ -1818,6 +1833,25 @@ contexts:
         - include: func-var
         - include: number
 
+  func-clamp:
+    - match: \b(clamp)(\()
+      captures:
+        1: support.function.clamp.css
+        2: punctuation.section.function.begin.css
+      push:
+        - meta_scope: meta.function.clamp.css
+        - match: '\('
+          push:
+            - match: '\)'
+              pop: true
+            - include: value-clamp
+        - include: size
+        - include: flex-non-negative
+        - include: percentage-non-negative
+        - include: length-non-negative
+        - include: end-func
+        - include: func-var
```

Comment:
1. `clamp(` pushes the contexts on the stack. Do we need to match another set of nested parens? If not, we can simplify this to the following:
```yaml
func-clamp:
   - match: \b(clamp)(\()
     captures:
       1: support.function.clamp.css
       2: punctuation.section.function.begin.css
     push:
       - meta_scope: meta.function.clamp.css
       - include: end-func
       - include: size
       - include: flex-non-negative
       - include: percentage-non-negative
       - include: length-non-negative
       - include: func-var
```
`end-func` will match the closing paren and pop all the contexts off the stack.

BTW It's a good practice to put the `pop` context at the top of the list. That way we pop as quickly as possible, reducing the size of the stack. This plugin has [a serious problem: it puts way too many contexts on the stack](https://github.com/ryboe/CSS3/issues/131). That makes it hard to embed in other syntax definitions like HTML and PHP without hitting the 25,000 context limit. This problem is the result of a conscious design choice I made when I wrote CSS3. It made sense at the time, but it has aged poorly.

2. I don't see `value-clamp` defined anywhere.

This also applies to `func-max` and `func-min`.

Reply:
I love learning all this, thanks for sharing! 

1. `clamp()` can be nested yeah, so can `min()` and `max()`. I copy and pasted from `calc()`, so def help me shape this better. I'll add a nesting example to `property-list.css`.
2. `value-clamp` good call, I'll add entries for `value-clamp`, `value-max`, and `value-min` 👍🏻

Reply:
So i removed the use of `value-clamp`, it looked like it would just duplicate `value-calc`, so i used `value-calc` and it seems to be working fine. thoughts? 

<img width="787" alt="Screen Shot 2021-02-22 at 3 53 13 PM" src="https://user-images.githubusercontent.com/1134620/108785347-149d6b00-7526-11eb-9a3f-97982e3f0ace.png">